shared/utility/io/uart.cpp

```diff
@@ -37,7 +37,7 @@ namespace utility::io {
         }
     }
 
-    void uart::close() {
+    void uart::close() const {
```

Comment:
It's incorrect that this should be able to be a const function. Within this function, we should instead be setting `fd = -1` when we close.